eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -571,10 +633,22 @@ object Commitments {
         val fees = commitTxTotalCost(commitments1.remoteParams.dustLimit, reduced, commitments.commitmentFormat)
         val missing = reduced.toRemote.truncateToSatoshi - commitments1.localParams.channelReserve - fees
         if (missing < 0.sat) {
-          Left(CannotAffordFees(commitments.channelId, missing = -missing, reserve = commitments1.localParams.channelReserve, fees = fees))
-        } else {
-          Right(commitments1)
+          return Left(CannotAffordFees(commitments.channelId, missing = -missing, reserve = commitments1.localParams.channelReserve, fees = fees))
+        }
+
+        // if we would overflow our dust exposure with the new feerate, we reject this fee update
+        if (feeConf.feerateToleranceFor(commitments.remoteNodeId).closeOnUpdateFeeDustExposureOverflow) {
+          val maxDustExposure = feeConf.feerateToleranceFor(commitments.remoteNodeId).maxDustHtlcExposure
+          val dustExposureAfterFeeUpdate = Seq(
+            CommitmentSpec.dustExposure(commitments1.localCommit.spec, fee.feeratePerKw, commitments1.localParams.dustLimit, commitments1.commitmentFormat),
+            CommitmentSpec.dustExposure(reduced, fee.feeratePerKw, commitments1.remoteParams.dustLimit, commitments1.commitmentFormat),
+          ).max
```

Comment:
Should it be `sum()` instead of `max` ?

Reply:
No, it's really max here.
We compute the dust exposure in our local commit and their remote commit.
But only one of these transactions can confirm on-chain, so it wouldn't make sense to sum them, right?